### ISSUE TITLE
Tags not in list

### ISSUE THREAD

**Issue by Vasperous** [squidfunk/mkdocs-material#3167]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [ ] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

I recently updated to the latest version of insiders and went to build my documentation.  The build failed before processing the pages with the following error:

```
Traceback (most recent call last):
  File "/Users/vasperous/.pyenv/versions/3.8.5/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/__main__.py", line 177, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 54, in serve
    config = builder()
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 49, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/commands/build.py", line 287, in build
    nav = config['plugins'].run_event('nav', nav, config=config, files=files)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/plugins.py", line 102, in run_event
    result = method(item, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/material/plugins/tags/plugin.py", line 74, in on_nav
    end = nav.pages.index(page, start + 1)
ValueError: Page(title='Tags', url='/tags.html') is not in list
```

I checked my `mkdocs.yml` and my commit history, since my last `mkdocs build -c` nothing has changed in my project except for the update to my material mkdocs version.  I have created a smaller 'shell' of my current project (since it is very large) to try and find out what plugin/feature is likely causing the issue.  The problem is that it works on my smaller project with the same configuration etc., the only big difference is the number of pages/macros/variables I have.

The only way I can get it to build is to remove `tags_file: tags.md` from the mkdocs.yml.

Here is my setup at a high level:
`mkdocs.yml`
```
use_directory_urls: false

nav:
  - Home:
    - Introduction: index.md
    - Getting Started: getting-started.md
    - Tags: tags.md

plugins:
  - search
  - localsearch
  - git-revision-date-localized
  - tags:
      tags_file: tags.md
```

Here is the output of the `tree -L 1` of the docs folder:
```
├── about
├── api
├── assets
├── estate-management
├── getting-started.md
├── images
├── implementation
├── index.md
├── reference
├── release-notes
├── stylesheets
└── tags.md
```

### Expected behaviour

Successfully build the project.

### Actual behaviour

Build process exists since it cannot find the tags.html:
```
Traceback (most recent call last):
  File "/Users/vasperous/.pyenv/versions/3.8.5/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/__main__.py", line 177, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 54, in serve
    config = builder()
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 49, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/commands/build.py", line 287, in build
    nav = config['plugins'].run_event('nav', nav, config=config, files=files)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/plugins.py", line 102, in run_event
    result = method(item, **kwargs)
  File "/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/material/plugins/tags/plugin.py", line 74, in on_nav
    end = nav.pages.index(page, start + 1)
ValueError: Page(title='Tags', url='/tags.html') is not in list
```

### Steps to reproduce

N/A since I am not able to repro using a smaller project.

### Package versions

- Python: `Python 3.8.5`
- MkDocs: `mkdocs, version 1.2.3 from /Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs (Python 3.8)`
- Material: `Version: 7.3.6+insiders.3.1.5`

### Configuration

```yaml
site_name: 'Documentation 2021-10 v01'
site_description: 'API documentation and implementation guide for Test'
site_author: 'Test'
site_url: ''

# Copyright
copyright: 'Copyright &copy; 2021 Test'

# If serving from a website you would want 'true', for offline you want false
use_directory_urls: false

nav:
  - Home:
    - Introduction: index.md
    - Getting Started: getting-started.md
    - Tags: tags.md
  - API:
    # Pages...
  - Estate Management:
    # Pages...
  - Implementation Guide: 
    # Pages...
  - Reference Guide:
    # Pages...
  - Release Notes:
    # Pages...
  - About:
    - License: about/license.md

theme:
  name: 'material'
  custom_dir: theme
  palette:
    scheme: 'dark'
    primary: 'dark-primary'
    accent: 'dark-accent'
  font:
    text: 'Roboto'
    code: 'Roboto Mono'
  logo: 'images/logo.png'
  favicon: 'images/favicon.png'
  icon:
    admonition:
      note: octicons/tag-16
      abstract: octicons/checklist-16
      info: octicons/info-16
      tip: octicons/squirrel-16
      success: octicons/check-16
      question: octicons/question-16
      warning: octicons/alert-16
      failure: octicons/x-circle-16
      danger: octicons/zap-16
      bug: octicons/bug-16
      example: octicons/beaker-16
      quote: octicons/quote-16
  features:
    - content.tabs.link
    - navigation.indexes
    - navigation.instant
    - navigation.tabs
    - navigation.tabs.sticky
    - navigation.top
    - navigation.tracking
    - search.highlight
    - search.share
    - search.suggest
    - toc.autohide

extra:
  generator: false

extra_css:
  - stylesheets/extra.css
  - stylesheets/icons.css
  - stylesheets/zoom.css

extra_javascript:
  - assets/javascripts/image-zoom.js

markdown_extensions:
  - abbr
  - admonition
  - attr_list
  - def_list
  - footnotes
  - meta
  - pymdownx.details
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
  - pymdownx.highlight:
      use_pygments: true
  - pymdownx.inlinehilite
  - pymdownx.keys
  - pymdownx.snippets:
      base_path: ext_md
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format

  - pymdownx.tabbed
  - toc:
      permalink: '#'
      toc_depth: 5

plugins:
  - search
  - localsearch
  - git-revision-date-localized
  - tags:
      tags_file: tags.md
  - macros:
      include_yaml:
        - ext_yaml/api-desc-event.yaml
        - ext_yaml/api-desc-request.yaml
        - ext_yaml/api-desc-response.yaml
        - ext_yaml/api-value-event.yaml
        - ext_yaml/api-value-request.yaml
        - ext_yaml/api-value-response.yaml
        - ext_yaml/misc.yaml
```

### System information

- Operating system: Mac OS Monterey
- Browser: Chrome & Brave

**Comment by squidfunk** [comment]
Thanks for reporting! Unfortunately, I somehow need a reproduction to debug it. Is the project available somewhere?

**Comment by Vasperous** [comment]
So I have played around with it a bit more and worked around the issue but I have still have no idea what the source of the issue is.  I removed my reference to the ` - Tags: tags.md` and it started to work even though I kept the `tags_file: tags.md` there.  Albeit without the tags page. 

When I ran it in verbose mode I see the following:
```
INFO     -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
              - tags.md
              - tags.md
```

Not sure where it is getting the two separate files of tags.md from...  That gave me an idea though, I renamed my `tags.md` to `tag.md` in my navigation and in the `tags_file`.  Unfortunately that didn't work either.

As a last resort I tried moving the location of my Tags page to the very bottom of my mkdocs.yml which magically resolved the issue.  I have around 275 pages (and growing) pages so I am not sure if this is a timing issue but based on log it picked up the tags.md right away:

```
INFO     -  [macros] - Extra variables (config file): ['generator']
DEBUG    -  [macros] - Content of extra variables (config file): {'generator': False}
INFO     -  [macros] - Extra filters (module): ['pretty']
DEBUG    -  [macros] - Docs directory: /Users/vasperous/Dev_Projects/USI/USI-Documentation/docs
DEBUG    -  [macros] - Includes directory: /Users/vasperous/Dev_Projects/USI/USI-Documentation/docs
INFO     -  Cleaning site directory
DEBUG    -  Looking for translations for locale 'en'
DEBUG    -  No translations found here: '/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/mkdocs/templates/locales'
DEBUG    -  No translations found here: '/Users/vasperous/.pyenv/versions/3.8.5/lib/python3.8/site-packages/material/locales'
DEBUG    -  No translations found here: '/Users/vasperous/Dev_Projects/USI/USI-Documentation/theme/locales'
DEBUG    -  Reading markdown pages.
DEBUG    -  Reading: index.md
DEBUG    -  Reading: getting-started.md
DEBUG    -  Reading: tags.md
```

**Comment by squidfunk** [comment]
Thanks for taking the time to debug the issue.

> Not sure where it is getting the two separate files of tags.md from... That gave me an idea though, I renamed my tags.md to tag.md in my navigation and in the tags_file. Unfortunately that didn't work either.

This is due to a hack in order to trigger a second rendering of the tags index page __if it's not the last page__, because otherwise, tags from all subsequent pages would be missing in the index. As tags are defined in the Markdown of a page, and the tags index is rendered in Markdown, we need to render the tags index _after_ all pages have been processed. This is trivial when the tags index page is the last page, but needs the second pass (which is why you're seeing two instances) when this is not the case. It's the best (and most flexible) approach I could come up so far, but I'll check if we can somehow improve it to mitigate the weird issues.

Did you have a change to disable all other plugins to see whether the issue persists?

**Comment by Vasperous** [comment]
That is the weird thing, I basically cloned by project and removed just about every page except the first three pages which includes my `tags.md`.  No changes to the plugins etc. and the project builds without issue. 

What I found was that this nav structure worked without issue:
```yaml
nav:
  - Home:
    - Introduction: index.md
    - Getting Started: getting-started.md
    - Tags: tags.md
```

But, the second I add something after the tags it breaks with the same error originally posted:
```yaml
nav:
  - Home:
    - Introduction: index.md
    - Getting Started: getting-started.md
    - Tags: tags.md
  - About: about/license.md
```

Moving to this structure also worked:
```yaml
nav:
  - Home:
    - Introduction: index.md
    - Getting Started: getting-started.md
  - About: about/license.md
  - Tags: tags.md
```

For now I am just going to put the tags at the end until I can find something more conclusive.  Thanks for your help! :)

**Comment by squidfunk** [comment]
Okay, I think I've got a fix in [`1a7f20889`](https://github.com/squidfunk/mkdocs-material-insiders/commit/1a7f208890444fcf5bfb7f8b74c4f42c3bd42a72). I could simplify the code and I've tested it with and without tags file, and with and without specifying the navigation. I hope that this covers all cases now.

**Comment by Vasperous** [comment]
Awesome thank you!

**Comment by squidfunk** [comment]
Released as part of 7.3.6-insiders-3.2.1.
